src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/DebugFuncEval.cs

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Reflection.Runtime.General;
+
+using Internal.TypeSystem;
+using Internal.Runtime.Augments;
+using Internal.Runtime.CallInterceptor;
+using Internal.TypeSystem.NativeFormat;
+using Internal.NativeFormat;
+
+namespace System.Runtime.InteropServices
+{
+    [McgIntrinsics]
+    internal static class AddrofIntrinsics
+    {
+        // This method is implemented elsewhere in the toolchain
+        internal static IntPtr AddrOf<T>(T ftn) { throw new PlatformNotSupportedException(); }
+    }
+
+    internal class AutoPinner : IDisposable
+    {
+        private GCHandle _pinnedArray;
+
+        public AutoPinner(Object obj)
+        {
+            _pinnedArray = GCHandle.Alloc(obj, GCHandleType.Pinned);
+        }
+
+        public static implicit operator IntPtr(AutoPinner ap)
+        {
+            return ap._pinnedArray.AddrOfPinnedObject();
+        }
+        
+        public void Dispose()
+        {
+            _pinnedArray.Free();
+        }
+    }
+}
+
+namespace Internal.Runtime.TypeLoader
+{
+    internal class DebugFuncEval
+    {
+
+        private static void HighLevelDebugFuncEvalHelperWithVariables(ref int param, ref LocalVariableSet arguments)
+        {
+            // Hard coding the argument integer here!
+            arguments.SetVar<int>(1, param);
+
+            // For now, fill this variable in the debugger, any static void method with a single integer argument would work
+            IntPtr targetAddress = RuntimeImports.RhpGetFuncEvalTargetAddress();
+
+            // Hard coding a single void return here
+            LocalVariableType[] returnAndArgumentTypes = new LocalVariableType[2];
+            returnAndArgumentTypes[0] = new LocalVariableType(typeof(void).TypeHandle, false, false);
+            returnAndArgumentTypes[1] = new LocalVariableType(typeof(int).TypeHandle, false, false);
+
+            // 2) Invoke the target method
+
+            // Hard coding static here
+            DynamicCallSignature dynamicCallSignature = new DynamicCallSignature(Internal.Runtime.CallConverter.CallingConvention.ManagedStatic, returnAndArgumentTypes, returnAndArgumentTypes.Length);
+
+            // Go and execute!
+            Internal.Runtime.CallInterceptor.CallInterceptor.MakeDynamicCall(targetAddress, dynamicCallSignature, arguments);
```

Comment:
Style: there's a `using Internal.Runtime.CallInterceptor` at the top. Do we really need to fully qualify?